crates/biome_grit_parser/src/parser/definitions.rs

```diff
@@ -86,7 +86,27 @@ fn parse_function_definition(p: &mut GritParser) -> ParsedSyntax {
     p.expect(T!['(']);
     VariableList.parse_list(p);
     p.expect(T![')']);
-    parse_curly_predicate_list(p).ok();
+
+    if p.at(JS_KW) {
+        p.bump(JS_KW);
+        p.expect(T!['{']);
+        // Comments inside the JS are not allowed
+        let mut brace_count = 0;
```

Comment:
If comments aren't allowed inside JS functions, maybe we should create a new lexer context, where we raise a parsing diagnostic every time we encounter a comment. At the moment it doesn't seem we do

Reply:
How do we handle other embedded languages? Defining an entire lexer for JS inside GritQL seems like overkill / problematic.

Reply:
We have a somewhat plan do that, but at the moment we don't have the infrastructure to do so. We plan to tackle the problem as part of the HTML parser, so once we solve it, we would be able to apply the solution to Grit too.